### PR TITLE
Update runtime

### PR DESCRIPTION
**Pull request by ruslandoga**
This PR updates Erlang/Elixir/Alpine versions to the current latest.

Context: https://github.com/hexpm/hexdocs/pull/44#discussion_r1806117683

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ COPY rel rel
 RUN mix release
 
 # prepare release image
-FROM alpine:3.17.3 AS app
+FROM alpine:3.20.3 AS app
 RUN apk add --no-cache --update bash openssl libstdc++
```
Comment (ruslandoga) [inline Dockerfile]: Should we also install `ca-certificates` (for the new `public_key.cacerts_get`) and `ncurses` (I never learned why it is needed in Erlang, but I think my apps fail to start when it's missing)?

File: mix.exs [hunk 2]
```diff
@@ -42,7 +42,8 @@ defmodule Hexdocs.MixProject do
   defp releases() do
     [
       hexdocs: [
-        include_executables_for: [:unix]
+        include_executables_for: [:unix],
+        reboot_system_after_config: true
```
Comment (ruslandoga) [inline mix.exs:46]: This preserves config/releases.exs behavior: https://hexdocs.pm/mix/Mix.Tasks.Release.html#module-config-providers

**Comment by ericmj** [comment]
There's a CI failure related to the config, I think you have to rename it to `runtime.exs`.

**Comment by ruslandoga** [comment]
Hm... I have renamed it.

The CI error

```
** (System.EnvError) could not fetch environment variable "HEXDOCS_PORT" because it is not set
    (elixir 1.17.3) lib/system.ex:705: System.fetch_env!/1
    /home/runner/work/hexdocs/hexdocs/config/runtime.exs:4: (file)
    (stdlib 6.1.2) erl_eval.erl:904: :erl_eval.do_apply/7
    (stdlib 6.1.2) erl_eval.erl:1192: :erl_eval.expr_list/7
    (stdlib 6.1.2) erl_eval.erl:446: :erl_eval.expr/6
    (stdlib 6.1.2) erl_eval.erl:436: :erl_eval.expr/6
    (stdlib 6.1.2) erl_eval.erl:1192: :erl_eval.expr_list/7
```

seems like a permissions issue.

**Comment by wojtekmach** [comment]
releases.exs was only evaluated in releases, runtime.exs is evaluated across the board including when we run tests. Adding `if config_env() == :prod` inside the file should fix it.

**Comment by ruslandoga** [comment]
@wojtekmach thank you!

**Review by ericmj — APPROVED** [review]

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**